### ISSUE TITLE
On Alpha versions, espeak pitch sometimes increases unexpectedly

### ISSUE THREAD

**Issue by leonardder** [nvaccess/nvda#10214]
### Steps to reproduce:
No reliable str. It seems to pop up after a moment where uppercase letters have been read.

### Actual behavior:
The overall pitch of espeak seems to increase. This isn't a natural pitch increase, but sounds more like the audio produced by espeak is played on a higher sample rate.

### Expected behavior:
pitch will be constant.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-18604

#### Windows version:
Windows 10 Version 1903 (OS Build 18970.1005)

#### Name and version of other software in use when reproducing the issue:
N/a

#### Other information about your system:
n/a

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Seems to be in all recent Alpha versions.

**Comment by XLTechie** [comment]
Can you resolve this by typing a Capital followed by a lowercase letter?

If so, that is a problem I know well from Linux versions of eSpeak, going back 
at least as far as 1.43.03.

**Comment by Brian1Gaff** [comment]
Yes sometimes sounds like a yodel. I noted this happens  a bit on other 
screenreaders if espeak is installed as sapi 5 though, and sometimes it goes 
the other way, into monotone, so are we sure this is nvda or Espeak?
 I suppose it could be both or either!

Brian

bglists@blueyonder.co.uk
Sent via blueyonder.
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users

**Comment by codeofdusk** [comment]
I've seen this too, post speech refactor.

**Comment by leonardder** [comment]
> 
> 
> I've seen this too, post speech refactor.

I'm thus not sure whether this is post speech refactor or also occurred before. Honestly, I always used Nuance Vocalizer pre speech refactor and switched to Espeak mainly due to speech refactor.

**Comment by leonardder** [comment]
Ugh, github reordered the close and comment/comment buttons. I"m sorry.

**Comment by leonardder** [comment]
> Can you resolve this by typing a Capital followed by a lowercase letter?

Yes.

> If so, that is a problem I know well from Linux versions of eSpeak, going back at least as far as 1.43.03.

Thanks for pointing this out. I will label this needs external fix.

**Comment by alex19EP** [comment]
i think espeak-ng/espeak-ng#975 is upstream issue for this.

**Comment by seanbudd** [comment]
is this still an issue @leonardder given https://github.com/espeak-ng/espeak-ng/issues/975 has been closed and integrated into NVDA?

**Comment by leonardder** [comment]
I'm afraid it is, though I haven't tested extensively with the newest alpha builds. I will report back later this week.
Note that it's strictly spoken not an issue with pitch, but more likely an issue with sample rate, i.e. it sounds like the audio is playing too fast.

**Comment by seanbudd** [comment]
is your inclination that it is still an issue with eSpeak or NVDA? seems like this bug will be hard to hunt down

**Comment by seanbudd** [comment]
That comment was from several years ago, @XLTechie, do you know if this is still an issue with the linux version of eSpeak?
I would suggest re-raising this issue with eSpeak then.

**Comment by leonardder** [comment]
I experienced the issue this very morning so it's definitely not fixed yet.

**Comment by Adriani90** [comment]
@leonardder the referenced upstream issue from eSpeak above is solved here:
https://github.com/espeak-ng/espeak-ng/pull/1515

I suspect this should work now in NVDA without problems right? At least after eSpeak submodule has been updated I think in NVDA 2023.1.

**Comment by leonardder** [comment]
I've still been able to repro this very recently.

**Comment by Adriani90** [comment]
I wonder whether this has something to do with the libsonic dependency used in eSpeak?

**Comment by codeofdusk** [comment]
I've seen this issue for years, and suspect it has something to do with improperly closed SSML as playing a capital followed by lower-case letter resets the pitch.

**Comment by Adriani90** [comment]
@alex19EP do you have maybe any thoughts to this issue? Can it be fixed by eSpeak at all? Or do we need to do anything from NVDA side?

**Comment by seanbudd** [comment]
oops - meant to close #15221 instead. I think a similar approach to #19661 could fix this

**Comment by Keithcat1** [comment]
Yeah, get these daily ever since I raised speech rate above 18 with rate boost on, if I'm not mistaken. Triggering playback of a capital letter makes it go away.
